src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -1168,6 +1169,33 @@ namespace ngraph
                         set_default_layouts(external_function, node);
                     }
                 }
+
+                template <>
+                void CPULayout::LAYOUT_DECL(ngraph::op::Concat)
+                {
+                    if (runtime::cpu::mkldnn_utils::use_mkldnn_kernel(node.get()))
+                    {
+                        auto concat = static_cast<const ngraph::op::Concat*>(node.get());
+                        auto input0_layout =
+                            runtime::cpu::mkldnn_utils::get_input_mkldnn_format(node.get(), 0);
+                        size_t num_inputs = node->get_input_size();
+
+                        vector<memory::format> prim_input_formats;
+                        vector<memory::format> prim_output_formats;
+                        for (size_t i = 0; i < num_inputs; i++)
+                        {
+                            prim_input_formats.push_back(input0_layout);
+                        }
+                        prim_output_formats.push_back(input0_layout);
```

Comment:
can you check with MKLDNN folks on what the output format should be? http://intel.github.io/mkl-dnn/group__c__api__concat.html Their API says we can either set output format/desc to NULL or create the primitive with mkldnn::any and query for the right output format.

Reply:
sure.

Reply:
@jbobba evarist said we can do either. Have any and query the output format or have the user define the output format. So what do you think is our way forward? Also, i've added a condition to allow only 2D and 4D inputs to use mkldnn concat, since its optimized for that. I have also made the other changes you and @pruthviIntel requested. Please take a look

Reply:
nitpick: I don't know if matters that much but should we make an effort and set the input layout to one of the biggest concat chunk(s) to minimize possible `ConvertLayout`s?

Reply:
evarist wanted us to query for output layouts. We need to create a primitive with 'any' as output layout and query that primitive for the actual layout.